packages/react/src/components/MapCard/MapControls.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { Layers16, CaretSortDown16 } from '@carbon/icons-react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+import { settings } from '../../constants/Settings';
+import { Button } from '../../index';
+import { MapControlPropType } from '../../constants/CardPropTypes';
+
+import ScrollingControls from './ScrollingControls';
+
+const { iotPrefix } = settings;
+const BASE_CLASS_NAME = `${iotPrefix}--map-controls`;
+
+const propTypes = {
+  /** list of map control buttons */
+  controls: PropTypes.arrayOf(MapControlPropType),
+  /** list of layered map control buttons that are expanded horizontally */
+  layeredControls: PropTypes.arrayOf(MapControlPropType),
+  i18n: PropTypes.shape({
+    layerTriggerIconDescription: PropTypes.string,
+    scrollUp: PropTypes.string,
+    scrollDown: PropTypes.string,
+  }),
+  /** true if the map controls are in an expanded card */
+  isExpandedMode: PropTypes.bool,
+  testId: PropTypes.string,
+  /** position of the control buttons tooltip */
+  tooltipPosition: PropTypes.oneOf(['left', 'top', 'bottom', 'right']),
+};
+const defaultProps = {
+  controls: [],
+  layeredControls: [],
+  i18n: {
+    layerTriggerIconDescription: 'Layered controls',
+    scrollUp: 'Scroll up',
+    scrollDown: 'Scroll down',
+  },
+  isExpandedMode: false,
+  testId: 'map-controls',
+  tooltipPosition: 'left',
+};
+
+const MapControls = ({
+  layeredControls,
+  controls,
+  tooltipPosition,
+  isExpandedMode,
+  i18n,
+  testId,
+}) => {
+  const { layerTriggerIconDescription, scrollUp, scrollDown } = i18n;
+  const [layersOpen, setLayersOpen] = React.useState(false);
+  const handleLayerClick = () => {
+    setLayersOpen((openState) => !openState);
+  };
+
+  const buttonSize = isExpandedMode ? 'field' : 'small';
+
+  const renderControl = (control, index) => {
+    if (control.group) {
+      const groupClass = `${BASE_CLASS_NAME}-btn-group`;
+      const groupKey = `btn-group-${index}`;
+      return control.hasScroll ? (
+        isExpandedMode ? (
+          <ScrollingControls
+            testId={`${testId}-scroll-controls`}
+            key={groupKey}
+            controls={control.group}
+            classname={groupClass}
+            scrollUpIconDescriptionText={scrollUp}
+            scrollDownIconDescriptionText={scrollDown}
+            tooltipPosition={tooltipPosition}
+            visibleItemsCount={control.visibleItemsCount}
+          />
+        ) : null
+      ) : (
+        <div key={groupKey} className={groupClass}>
+          {control.group.map((groupedControl, myIndex) => {
+            return (
+              <Button
+                key={`${groupedControl.iconDescription}-${myIndex}`}
```

Comment:
Should consider using something unique to the value rather than the index

Reply:
Yes, we are already using `groupedControl.iconDescription` as part of the key which should be unique, but as a backup we also append the index. I theory this can still go wrong but the risk of that happening is probably negligible.